### PR TITLE
core: bump packaging dep to v24

### PR DESCRIPTION
**Pull request by dokterbob**
Allow installing core with packaging@^24.

### CODE REVIEW

File: libs/core/pyproject.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ langsmith = "^0.1.0"
 tenacity = "^8.1.0"
 jsonpatch = "^1.33"
 PyYAML = ">=5.3"
-packaging = "^23.2"
+packaging = ">=23.2,^24.0"
```
Comment (eyurtsev) [inline libs/core/pyproject.toml:18]: >=23.2 implies 24 and 25 if i'm not mistaken?

Comment (dokterbob) [reply libs/core/pyproject.toml:18]: `>=` equal or higher than.

AFAIK this implies higher than or equal to 23.2 and lower than 25. 

Ref: https://python-poetry.org/docs/dependency-specification/

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
Thanks! We'll try to see if we can remove packaging as a dependency to avoid dependency conflicts for users

**Comment by dokterbob** [comment]
> Thanks! We'll try to see if we can remove packaging as a dependency to avoid dependency conflicts for users

That's great to hear but in the mean time, perhaps we could use my code to fix the issue for the time being?

Just rebased my PR to master.

File: libs/core/pyproject.toml (hunk 1 above)
Comment (baskaryan) [reply libs/core/pyproject.toml:18]: afaik the `,` indicates a logical AND not a logical OR, so `>=23.2,^24.0` ends up just being `^24.0`. should it isntead be `>=23.2,<25`?

**Comment by mbastian** [comment]
Hi folks, I would appreciate if we could merge that as my project currently has conflicts due to this library as another dependency has >=24.0. Thanks in advance!

**Comment by hwchase17** [comment]
fixed in other pr

**Pull request closed without merging**